admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ConfigurationEdit.view.js

```diff
@@ -121,7 +121,7 @@ define([
             _.each(this.collectionArray.models, function(model) {
                 values.push(model.get('value'));
             });
-            this.configuration.get('properties').set(this.model.get('id'), values);
+            this.configuration.get('properties').set(this.model.get('id'), values.length === 0 ? null : values);
```

Comment:
I was about to ask the same thing. I prefer empty arrays to null arrays personally. Seems like this change could cause more issues down the road.